Improve error message when deal.II found but not the right version.

One of my students (Jerett Cherry) ended up spending too much time when he ran into this error message that we have in all of our tutorial programs and code gallery programs:
```
FIND_PACKAGE(deal.II 9.4.0 QUIET
  HINTS ${deal.II_DIR} ${DEAL_II_DIR} ../ ../../ $ENV{DEAL_II_DIR}
  )
IF(NOT ${deal.II_FOUND})
  MESSAGE(FATAL_ERROR "\n"
    "*** Could not locate a (sufficiently recent) version of deal.II. ***\n\n"
    "You may want to either pass a flag -DDEAL_II_DIR=/path/to/deal.II to cmake\n"
    "or set an environment variable \"DEAL_II_DIR\" that contains this path."
    )
ENDIF()
```
The problem was that he had pointed `DEAL_II_DIR` at too old a version of deal.II. The error message is awkward because it does not discriminate between two cases one would like to understand when getting this error:
* No version of deal.II found at all
* A version was found, but it was too old.

Do we have a way to improve this situation? Does `FIND_PACKAGE` report anything useful if it finds a version of the package in question, but then realizes that it isn't new enough?

FWIW I am seeing
```
$ cmake ..
CMake Warning at CMakeLists.txt:26 (find_package):
  Could not find a configuration file for package "deal.II" that is
  compatible with requested version "10.0.0".

  The following configuration files were considered but not accepted:

    /tmp/Software/dealii_new/build/lib/cmake/deal.II/deal.IIConfig.cmake, version: 9.5.0



CMake Error at CMakeLists.txt:30 (message):
  

  *** Could not locate a (sufficiently recent) version of deal.II.  ***



  You may want to either pass a flag -DDEAL_II_DIR=/path/to/deal.II to cmake

  or set an environment variable "DEAL_II_DIR" that contains this path.


-- Configuring incomplete, errors occurred!
```
for
```
find_package(deal.II 10.0.0
  HINTS ${deal.II_DIR} ${DEAL_II_DIR} ../ ../../ $ENV{DEAL_II_DIR}
  )
if(NOT ${deal.II_FOUND})
  message(FATAL_ERROR "\n"
    "*** Could not locate a (sufficiently recent) version of deal.II. ***\n\n"
    "You may want to either pass a flag -DDEAL_II_DIR=/path/to/deal.II to cmake\n"
    "or set an environment variable \"DEAL_II_DIR\" that contains this path."
    )
endif()
```
, i.e,  if you omit `QUIET`. Most examples don't use `QUIET` anyway:
```
$ git grep -in "FIND_PACKAGE" ../.. | grep -i "QUIET"
../../step-19/CMakeLists.txt:26:find_package(deal.II 9.5.0 QUIET
../../step-66/CMakeLists.txt:26:find_package(deal.II 9.5.0 QUIET
../../step-68/CMakeLists.txt:26:find_package(deal.II 9.5.0 QUIET
../../step-71/CMakeLists.txt:26:find_package(deal.II 9.5.0 QUIET
../../step-72/CMakeLists.txt:26:find_package(deal.II 9.5.0 QUIET
```

--- a/examples/step-19/CMakeLists.txt
+++ b/examples/step-19/CMakeLists.txt
@@ -23,7 +23,7 @@ set(TARGET_SRC
 
 cmake_minimum_required(VERSION 3.13.4)
 
-find_package(deal.II 9.5.0 QUIET
+find_package(deal.II 9.5.0
   HINTS ${deal.II_DIR} ${DEAL_II_DIR} ../ ../../ $ENV{DEAL_II_DIR}
   )
 if(NOT ${deal.II_FOUND})

--- a/examples/step-66/CMakeLists.txt
+++ b/examples/step-66/CMakeLists.txt
@@ -23,7 +23,7 @@ set(TARGET_SRC
 
 cmake_minimum_required(VERSION 3.13.4)
 
-find_package(deal.II 9.5.0 QUIET
+find_package(deal.II 9.5.0
   HINTS ${deal.II_DIR} ${DEAL_II_DIR} ../ ../../ $ENV{DEAL_II_DIR}
   )
 if(NOT ${deal.II_FOUND})

--- a/examples/step-68/CMakeLists.txt
+++ b/examples/step-68/CMakeLists.txt
@@ -23,7 +23,7 @@ set(TARGET_SRC
 
 cmake_minimum_required(VERSION 3.13.4)
 
-find_package(deal.II 9.5.0 QUIET
+find_package(deal.II 9.5.0
   HINTS ${deal.II_DIR} ${DEAL_II_DIR} ../ ../../ $ENV{DEAL_II_DIR}
   )
 if(NOT ${deal.II_FOUND})

--- a/examples/step-71/CMakeLists.txt
+++ b/examples/step-71/CMakeLists.txt
@@ -23,7 +23,7 @@ set(TARGET_SRC
 
 cmake_minimum_required(VERSION 3.13.4)
 
-find_package(deal.II 9.5.0 QUIET
+find_package(deal.II 9.5.0
   HINTS ${deal.II_DIR} ${DEAL_II_DIR} ../ ../../ $ENV{DEAL_II_DIR}
   )
 if(NOT ${deal.II_FOUND})

--- a/examples/step-72/CMakeLists.txt
+++ b/examples/step-72/CMakeLists.txt
@@ -23,7 +23,7 @@ set(TARGET_SRC
 
 cmake_minimum_required(VERSION 3.13.4)
 
-find_package(deal.II 9.5.0 QUIET
+find_package(deal.II 9.5.0
   HINTS ${deal.II_DIR} ${DEAL_II_DIR} ../ ../../ $ENV{DEAL_II_DIR}
   )
 if(NOT ${deal.II_FOUND})

